flink-table/flink-table-common/src/main/java/org/apache/flink/table/api/TableSchema.java

```diff
@@ -346,22 +322,137 @@ public static Builder builder() {
 		return new Builder();
 	}
 
+	//~ Tools ------------------------------------------------------------------
+
+	/**
+	 * Tools method to transform arrays of table names and types
+	 * into a {@link TableColumn} list.
+	 */
+	private static List<TableColumn> createTableColumns(
```

Comment:
still don't feel we should introduce this method, 2 reasons:
1. it's only used in one place
2. it contains a validation logic which seems to be randomly IMO. Just copy the implementation to the caller would be more clear.

Reply:
How could i copy ? Call the `this()` must be the first statement in the constructor.